.github/workflows/docker-build.yml

```diff
@@ -0,0 +1,155 @@
+---
+name: Build-Publish
+
+on:
+  push:
+    branches:
+  release:
+    types: [released]
+
+jobs:
+  build-publish:
+    name: "Python-${{ matrix.PYTHON }}"
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        PYTHON:
+          - 3.7.16
+          - 3.8.16
+          - 3.9.16
+          - 3.10.9
+          - 3.11.1
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Set Variables
+        id: vars
+        run: |
+          # Extract python subversions
+          PYTHON_MAJOR=$(echo $PYTHON | sed 's/\.[0-9]\+$//')
+          PYTHON_PATCH=$(echo $PYTHON | sed 's/^[0-9]\+\.[0-9]\+\.//')
```

Comment:
It would have been more readable with
```
MAJOR="$( echo "${PYTHON}" | awk -F'.' '{print $1}' )"
MINOR="$( echo "${PYTHON}" | awk -F'.' '{print $2}' )"
PATCH="$( echo "${PYTHON}" | awk -F'.' '{print $3}' )"
```

But I'ma let this one slip